test/extended/networking/vlan.go

```diff
@@ -0,0 +1,134 @@
+package networking
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	nadtypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	corev1 "k8s.io/api/core/v1"
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	podframework "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-network][Feature:vlan]", func() {
+	oc := exutil.NewCLI("vlan")
+	f := oc.KubeFramework()
+	for _, pluginType := range []string{"vlan", "ipvlan", "macvlan"} {
```

Comment:
@mmirecki You can still use a for loop for the acutal testcase logic, but don't for-loop the setup. Do that with BeforeEach()/AfterEach() so that things get cleaned up correctly too.

Anything that's common between the tests can go into the BeforeEach() and be a variable declared inside the top-level Describe(). BeforeEach() should also generate a unique bridge name to pass to the CNI plugin. Be a bit careful here of parallel tests since each spec can run in parallel and variables sometimes get evaluated before the test is actually run; we can handle that in the resubmit PR review though. 

AfterEach() needs to clean up anything the test creates in the kube API, like all the attachment definitions if they exist, so that errors that stop the test in the middle don't leave them around.